docs/digitalocean.rst

```diff
@@ -0,0 +1,178 @@
+
+
+Getting Wooey to talk to Nginx & uWSGI
+---------------------
+
+We will basically follow the tutorial [here](http://uwsgi-docs.readthedocs.io/en/latest/tutorials/Django_and_nginx.html). If you installed Wooey in a virtualenv from the get go, then follow all those steps, if not you can ignore that set:
+
+Then we can follow the guide along with a couple changes:
+
+1. Basic test
+
+    * Because I was having trouble using non-standard ports, I replaced 8000 with 80 and ran the commands as root
+
+2. Test your Django project
+
+    * In order to get this step to work correctly I found I had to call uWSGI from the main folder where manage.py lives, in my case that was /home/user/ProjectName (I have no idea why, but otherwise it won't find the Wooey project correctly)
+
+3. Deploying static files
+
+    * So far I have ignored this altogether and no problems...      
+
+4. nginx and uWSGI and test.py
+
+    * Again only got this to work on port 80 again because of non-standard port problems
+
+
+Now if you want to run on a network socket, at this point you should be good to go. (Remember crucially this needs to be run from the same folder as manage.py).
+
+::
+
+    uwsgi --socket 127.0.0.1:8000 --wsgi-file ProjectName/wsgi.py --chmod-socket=666
+
+If you want to use a file socket, I then created an empty file in the Wooey project directory to be used as one (in this example named django.sock).
+
+::
+
+    uwsgi --socket ProjectName/django.sock --wsgi-file ProjectName/test.py --chmod-socket=666
+
+
+At this point we should now have Wooey running on Port 80 through Nginx.
+
+Forcing SSL with Nginx
+---------------------
+
+I have forced SSL with the following settings. (I think I might be running two SSL redirects, one on the Nginx side and one on the Django side which is never necessary because Nginx comes first, any clarification would be welcome, however for those following along:)
+
+I switched the main nginx block to HTTPS (there's a good tutorial [here](https://www.digitalocean.com/community/tutorials/how-to-create-an-ssl-certificate-on-nginx-for-ubuntu-14-04) if you haven't done this before).
+
+I also added an HTTPS header to the server block listening on 443 so Django knows it's HTTPS:
+
+::
+
+    proxy_set_header X-Forwarded-Proto $scheme;
+
+Then I set-up a second server block to listen on port 80 and rewrite to https:
+
+::
+
+    server {
+        listen   80;
+        listen   [::]:80;
+
+        server_name  46.101.9.192;
+
+        return 301 https://$server_name$request_uri;
+    }
+
+
+Then on the Django side I added the following flags to my config in user_settings.py
+
+::
+
+    SECURE_SSL_REDIRECT = True #this may be the double redirect which is unnecessary.
+    SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
```

Comment:
I would note that this and the above are settings that should be untoggled:

https://github.com/wooey/Wooey/blob/master/wooey/conf/project_template/settings/user_settings.py#L77